### PR TITLE
Moving cannonkeys boards to one place, eeprom fix

### PR DESCRIPTION
**Pull request by awkannan**
## Description
Two changes rolled into this PR:
1. Moving boards that I maintain to their own folder, and have them share code
1. Fix on eeprom update to check that the data needs to be updated before writing

## Types of changes

- [X] Core
- [X] Bugfix
- [ ] New Feature
- [ ] Enhancement/Optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/Layout/Userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by this PR

* https://github.com/qmk/qmk_firmware/issues/4990

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document. (https://docs.qmk.fm/#/contributing)
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cannonkeys/ortho48/rules.mk [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # project specific files
+VPATH += keyboards/cannonkeys/bluepill
 SRC =	led.c \
-		  ws2812.c
+		  ws2812.c \
```
Comment (drashna) [inline keyboards/cannonkeys/ortho48/rules.mk]: If you're including this, you may want to make sure that you have `RGBLIGHT_CUSTOM_DRIVER = yes` here too.

File: keyboards/cannonkeys/ortho48/rules.mk [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # project specific files
+VPATH += keyboards/cannonkeys/bluepill
```
Comment (drashna) [inline keyboards/cannonkeys/ortho48/rules.mk]: Shouldn't need this, in theory. Should already be done

File: keyboards/cannonkeys/ortho48/rules.mk (hunk 1 above)
Comment (awkannan) [reply keyboards/cannonkeys/ortho48/rules.mk]: I actually want to rely on the default `rgblight_set` behavior, so I shouldn't define it.

The ws2812.c file acts as if its the driver for ws2812.

**Comment by drashna** [comment]
Thanks!

**Pull request merged**